tephra-core/src/main/java/co/cask/tephra/persist/HDFSTransactionLog.java

```diff
@@ -157,11 +224,69 @@ public TransactionEdit next(TransactionEdit reuse) throws IOException {
       if (closed) {
         return null;
       }
-      boolean successful = reader.next(key, reuse);
-      if (successful) {
-        return reuse;
+      if (transactionEdits.size() != 0) {
+        return transactionEdits.remove(0);
+      } else {
+        if (newVersion) {
```

Comment:
Instead of calling it `newVersion` as a boolean, better have an enum to represent the version.